Provide io streams as plain values and set args via settings

bluurryy opened the pull request:
Fixes #522

Of note:
- **io streams are initialized to an empty, dummy file that does nothing on write**
  - i've implemented `KotoFile` for [`std::io::Empty`](https://doc.rust-lang.org/std/io/struct.Empty.html) to represent this dummy implementation
  - this is only observable when manually creating the modules via `make_module` or `CoreLib::default`
- **the default value of `KotoVmSettings::args` is `vec![]`**
  - Given that the io streams' default values are the process's io streams, it could also make sense to initialize the args to the process's... However, `std::env::args()` can panic, which I'd rather not have in a default implementation. This would raise the question of how to handle non-utf8 args. I think it would be better to have a `KotoSettings::inherit_args()` method that uses `std::env::args()` if anything. Would it be better if the io streams were dummy initialized with an "empty" file and require `KotoSettings::inherit_io()` for the current behavior?

irh reviewed (commented):
> Given that the io streams' default values are the process's io streams, it could also make sense to initialize the args to the process's...

You have a point here, and it would probably be worth making it opt-in for apps that embed Koto to inherit both args and IO, and `KotoSettings::inherit_args()` and `inherit_io()` would be useful helpers.

I'm not sure about the default for the io streams being an empty file though, I would prefer to avoid silent failures. The use case I have in mind is for embedded scripting where the application doesn't want the script to be accessing IO. If a script _does_ attempt to access IO then they should get an error. The default `File` "unsupported for this file type" error message isn't very helpful though, something like `stdin is unavailable` would be better.

bluurryy commented:
> You have a point here, and it would probably be worth making it opt-in for apps that embed Koto to inherit both args and IO, and `KotoSettings::inherit_args()` and `inherit_io()` would be useful helpers.

I've implemented this here.

> I'm not sure about the default for the io streams being an empty file though, I would prefer to avoid silent failures.

Oh yeah I like that. I've added `UnavailableStdin`, `UnavailableStdout` and `UnavailableStderr` as the default io stream file implementations.

The existing `Default*` io stream types would now have a misleading name in my opinion so I've renamed them to `System*`.
